tests/unit/Metrics/Transports/OpenTelemetryMetricTransportFactoryTest.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopware\OpenTelemetry\Tests\Unit\Metrics\Transports;
+
+use OpenTelemetry\API\Metrics\MeterInterface;
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\TestCase;
+use Shopware\Core\Framework\Telemetry\Metrics\Config\TransportConfig;
+use Shopware\Core\Framework\Telemetry\Metrics\Metric\Type;
+use Shopware\OpenTelemetry\Feature;
+use Shopware\OpenTelemetry\Metrics\MetricNameFormatter;
+use Shopware\OpenTelemetry\Metrics\Transports\OpenTelemetryMetricTransport;
+use Shopware\OpenTelemetry\Metrics\Transports\OpenTelemetryMetricTransportFactory;
+use Shopware\OpenTelemetry\Metrics\Transports\OpenTelemetryMeterProviderFactory;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use OpenTelemetry\API\Metrics\MeterProviderInterface;
+
+#[CoversClass(OpenTelemetryMetricTransportFactory::class)]
+class OpenTelemetryMetricTransportFactoryTest extends TestCase
+{
+    public function setUp(): void
+    {
+        if (!Feature::metricsSupported()) {
+            static::markTestSkipped('Installed version of shopware/core does not support metrics');
+        }
+    }
+
+    public function testCreate(): void
+    {
+
+        $namespace = 'testNamespace';
+        $formatter = $this->createMock(MetricNameFormatter::class);
+        $meterProviderFactory = $this->createMock(OpenTelemetryMeterProviderFactory::class);
+        $container = $this->createMock(ContainerInterface::class);
+        $meterProvider = $this->createMock(MeterProviderInterface::class);
+
+        $transportConfig = new TransportConfig([
+            (object) [
```

Comment:
Using object workaround here, as mocking MetricConfig is problematic because of private constructor and read-only class.
Using MetricConfig factory method looks like not a good idea, as it may change when yaml config format is changed. At the same time MetricConfig public properties is a part of the public API for transports.

Reply:
I think if constructor should stay final then `fromDefinition` should be treated as public api - can always add extra static constructors

Reply:
fromDefinition is less stable, as it used to create MetricConfig from array received from config. I would either make constructor public, or remove readonly from class (to make it possible to mock).

Reply:
as you wish, I don't mind :)

Reply:
It's clearly a bit of a pain to test. Just for context, the aim of the final design, is that this object should strictly be created from inside platform.  Externals can get it, but never initialize it.  (which translates to having fromDefinition @internal and the constructor private, but the object itself is the public API).

With that in mind, the annoyance here is that it's annoying to use in tests. (mock or initialize)

The reason for this change is totally valid, but also the compromise is annoying

Reply:
can we subclass it in the test namespace of this project? To make it easier to use in tests?

but just thinking about it, probably i would also say that from definition can be treated as public api, as that schema is basically what we expose as configuration, so when we remove a param there etc it would not only break the tests here but also the production config of the shops out there that made use of that right?

Reply:
`MetricConfig` properties are public interface. So we agreed that we will make constructor also public, as it will just have all properties. This way we will be able to initialize configs in tests.
Factory method `fromDefinition` that creates object from configuration array will stay private, as it may change when yaml configuration format changes.

@keulinho I hope you agree.

Reply:
Good point, agree that `fromDefinition` is public as it's tied to configurations of production shop.
But transports for it's work are not dependent on it. So if we make it a part of interface with tests (remove @internal annotation) and use it in tests, we would add another reason to change transport interface, and this way would make it less stable.

As a compromise at the moment we've made `MetricConfig` constructor public.